From 0887ce0d18d4b437b3b17e3a79f83c04b1234ffc Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Mon, 21 May 2012 16:34:33 +0800
Subject: [PATCH 204/280] ENGR00210129 Bugfix: (i.MX6) All the Mjpeg clips
 can't seek, after seek, all the clips can't play

Phenomenon:
For i.MX6, All the Mjpeg clips can't seek, after seek, all the clips
can't play.

Root cause:
The seek API vpu_DecBitBufferFlush will issue DEC_BUF_FLUSH command to
VPU FW. But i.MX6 MJPEG doesn't need FW though there exists MJPEG FW
code. In FW v2.1.4, the MJPEG FW code related with host command was
removed, thus if VPU lib issues host command, there will be undefined
behaviour.

Solution:
For i.MX6 MJPEG, VPU lib will not issue DEC_BUF_FLUSH command (actually
will not issue any host command).

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 vpu/vpu_lib.c | 6 ++++--
 vpu/vpu_lib.h | 2 +-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index a6e145c..1440856 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -4374,8 +4374,10 @@ RetCode vpu_DecBitBufferFlush(DecHandle handle)
 	if (cpu_is_mx6x())
 		pCodecInst->ctxRegs[CTX_BIT_RD_PTR] = pDecInfo->streamBufStartAddr;
 
-	BitIssueCommand(pCodecInst, DEC_BUF_FLUSH);
-	while (VpuReadReg(BIT_BUSY_FLAG)) ;
+	if (!is_mx6x_mjpg_codec(pCodecInst->codecMode)) {
+        BitIssueCommand(pCodecInst, DEC_BUF_FLUSH);
+        while (VpuReadReg(BIT_BUSY_FLAG)) ;
+    }
 
 	pDecInfo->streamWrPtr = pDecInfo->streamBufStartAddr;
 
diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index a42452b..0eb9f8e 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -667,7 +667,7 @@ typedef struct vpu_versioninfo {
  * v4.2.2 [2008.09.03] support encoder on MX51
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 1)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 2)
 
 extern unsigned int system_rev;
 
-- 
1.8.0

